generated/nifake/nifake/unit_tests/test_session.py

```diff
@@ -966,6 +966,25 @@ def test_set_attribute_string(self):
             session.read_write_string = attrib_string
             self.patched_library.niFake_SetAttributeViString.assert_called_once_with(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), _matchers.ViStringMatcher(''), _matchers.ViAttrMatcher(attribute_id), _matchers.ViStringMatcher('This is test string'))
 
+    def test_get_attribute_string_with_converter(self):
+        self.patched_library.niFake_GetAttributeViString.side_effect = self.side_effects_helper.niFake_GetAttributeViString
+        string = 'not that interesting'
+        self.side_effects_helper['GetAttributeViString']['attributeValue'] = string
+        with nifake.Session('dev1') as session:
+            attr_string = session.read_write_string_repeated_capability
+            assert attr_string == string
+            from mock import call
+            calls = [call(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), _matchers.ViStringMatcher(''), _matchers.ViAttrMatcher(1000010), _matchers.ViInt32Matcher(0), None), call(_matchers.ViSessionMatcher(SESSION_NUM_FOR_TEST), _matchers.ViStringMatcher(''), _matchers.ViAttrMatcher(1000010), _matchers.ViInt32Matcher(20), _matchers.ViCharBufferMatcher(len(string)))]
+            self.patched_library.niFake_GetAttributeViString.assert_has_calls(calls)
+            assert self.patched_library.niFake_GetAttributeViString.call_count == 2
+
+    def test_set_attribute_string_with_converter(self):
+        self.patched_library.niFake_SetAttributeViString.side_effect = self.side_effects_helper.niFake_SetAttributeViString
+        attribute_id = 1000010
+        with nifake.Session('dev1') as session:
+            session.read_write_string_repeated_capability = 42
```

Comment:
I think the `nifake` test covering a single conversion type is sufficient - since the converter function is heavily tested. However it would be best if you can mock the conversion function, then we know for sure we're good.

Is this a reasonable thing to do?

Reply:
I added this unit test to keep the coverage up. So the intent is to just exercise the API method.

What would mocking the converter get us?

Reply:
>What would mocking the converter get us?
 
Our Python APIs accept all sorts of things for repeated capabilities:
* `0`
* `[0,1]`
* `"0,1"`
* etc... 

We know the conversion is done by a single function so this test only tests one type of conversion. In essence, the test assumes that if one conversion worked, all of them will because they go through the one function. Which is true today but there is guarantee that it will be true in the future.

So we could either stop assuming it goes through that one function by checking for it explicitly through a mock (test the implementation, not great) or we can test more types (test the behavior, arguably more correct).

Reply:
> test assumes that if one conversion worked, all of them will because they go through the one function. Which is true today but there is guarantee that it will be true in the future.

I think we can or should trust ourselves to not break this in the future.

As you mentioned, mocking isn't great since we'll testing the implementation. Testing more types is better but this is not the right place to do it. Unit tests for `_converters.convert_repeated_capabilities()` is the more appropriate place and we do test all the supported types there.

I'm inclined to the leave the tests as is, unless you feel strongly against it.